### ISSUE TITLE
[1.10.2] move block event?

### ISSUE THREAD

**Issue by FiveEYZ** [SpongePowered/SpongeForge#799]
RFTools Elevator as seen on the picture will move to the desired level but also stay on current and hence duplicating the blocks it uses as floor.

https://gyazo.com/7cca4f9e8bfae05df72bef4489f0fafb

this all started when we enabled sponge on the server
nothing shows in the logs and 

current sponge installed is:
spongeforge-1.10.2-2026-5.0.0-BETA-1626

**Comment by gabizou** [comment]
Can you list instructions on how to replicate this exactly? Seeing as I've never played with RFTools, I have little to no clue how to get that sort of thing started.

**Comment by inklit** [comment]
I've seen a very similar bug with OpenComputer's Robots on SpongeForge 1.8.9.
The robot leaves a trail of itself behind.

**Comment by gabizou** [comment]
Yes, again, I have not had the experience of using any of those mods, so if you can please provide a list of instructions on replicating this, that'd be great.

**Comment by inklit** [comment]
1: Install Forge 1902, Sponge 1625,  and OpenComputers 1.5.22
2: In your creative menu search for "Creatix" (This is a prebuilt Robot for creative users and debugging purposes)
3: Place down "Creatix" and right click the Robot.
4: Press the Power button on the Robot to turn it on. You will then be prompted with a operating system of sorts that can be programmed via Lua.
5: Type "lua" to start the Lua interpreter.
6: Type "robot.forward()" and press Enter.
It will appear that nothing has happened, until you update the block in front of the robot by either right clicking it or relogging.
There, you will now find a duplicated Robot.

**Comment by FiveEYZ** [comment]
was in middle of making a tutorial but i did it on a server with only sponge and rftools installed

when i tested the elevator this happen: https://gist.github.com/FiveEYZ/c77f4db5dd777bc6fdc698040cf868d3

will now try the same with opencomputer

**Comment by ryantheleach** [comment]
I assume this is due to moving tile entities?

**Comment by inklit** [comment]
That is right @ryantheleach

**Comment by ryantheleach** [comment]
Does computercraft also use moving tile entities? or does it use an entity as well as a tile entity?

**Comment by inklit** [comment]
@ryantheleach I'm not sure, it's closed source. The only person that I know that might know the answer to that is @SquidDev-CC

**Comment by FiveEYZ** [comment]
dont know what u did, but its fixed with Build 1646 thanks a bunch

**Comment by FiveEYZ** [comment]
dont know what really changed, but this is sadly back
https://gist.github.com/FiveEYZ/e375341a15bd3141f4386e6d77f21635

tested with:
forge-1.10.2-12.18.1.2045-universal
spongeforge-1.10.2-2044-5.0.0-BETA-1655
mcjtylib-1.10-1.9.9
rftools-1.10-5.10

**Comment by inklit** [comment]
It was never fixed to begin with.
OpenComputers still experiences the same issue. This is not exclusive to 1.10.2 by the way. It also affects 1.8.9.

**Comment by Eufranio** [comment]
Yeah it happens here too. I think ComputerCraft handles turtles moving in a different way, OC and RFTools are more generic... Any news?

**Comment by Nentify** [comment]
Same issue with OpenComputers here, hopefully this can be fixed.

Screenshot:

![](http://puu.sh/qyPLt/eaf72bd7ed.jpg/ss%20(2016-08-12%20at%2004.45.38).jpg)

**Comment by inklit** [comment]
@gabizou I thought I'd mention this again, since you labeled it under 1.10, it is actually not exclusive to 1.10.
The bug affects 1.8.9 as well.

**Comment by bloodmc** [comment]
@Eufranio @inklit is this still an issue with latest SF 1.10.2 builds?

**Comment by Eufranio** [comment]
Not here.

**Comment by Aaron1011** [comment]
Oops, sorry.

**Comment by inklit** [comment]
I can't test with 1.10.2, but it is still an issue for 1.8.9.

**Comment by Eufranio** [comment]
Elevators from RFTools are working here, but Robots from OpenComputers are duped. Steps to reproduce:
1. Place a Robot (I'm using Creatix):
   ![cda5481a1295428cb090c5a9b9b746d3](https://cloud.githubusercontent.com/assets/17598280/19786325/2d26b546-9c7c-11e6-9586-1f2b590c4ebf.png)
2. Open, turn on and enter in the LUA terminal using `lua` command:
   ![cda5481a1295428cb090c5a9b9b746d3](http://image.prntscr.com/image/180b97c078414064954d2091b3e21a32.png)
3. Send a move comand to the Robot, like `robot.forward()`. Visually, the Robot won't move, and Sponge will show a error in the console **if the robot is inside a claim**:
   ![cda5481a1295428cb090c5a9b9b746d3](http://image.prntscr.com/image/56f878b5f5a2479ab3bbb9e72af63a03.png)
   `[Sponge]: No entity type is registered for class li.cil.oc.server.agent.Player`
4. If you update the block in the front of the Robot, you will see the dupped Robot (but probably without any NBT data):
   ![cda5481a1295428cb090c5a9b9b746d3](http://image.prntscr.com/image/b746ec2b4ce94ee083ec1e6c0933b3c5.png)

We can't open the dupped Robot (probably because of this (https://github.com/SpongePowered/SpongeForge/commit/d7287a150a477da190e1d78938715047bfa77609) commit, we could open the robot before). As you see, 2 Robots.

You can also break the Robots, they will drop. This bug doesn't happens with ComputerCraft or RFTools.

**Comment by bloodmc** [comment]
@Eufranio Please test again with latest SF and GP.

**Comment by FiveEYZ** [comment]
only 1 issue remain with RFTools elevator the, the first issue where the blocks would get duped is gone.

Instead now it will leave a "ghost" block to the client, if u destroy it it wont change for other clients looking at the same block/s and the block/s wont drop (unless you destroy a block in the elevator at the floor its currently at).

**Comment by ghost** [comment]
Have same issue with SpongeForge and OpenComputers robots on 1.10.2 - build 1879. (No GP installed).

**Comment by LizzyTrickster** [comment]
Paging @fnuecke

**Comment by ghost** [comment]
OC Robot moving:

https://github.com/MightyPirates/OpenComputers/blob/OC1.6-MC1.10/src/main/scala/li/cil/oc/common/tileentity/Robot.scala#L189

**Comment by ghost** [comment]
When turn on sponge debug:

Place robot:

<pre>
[14:32:16] [Server thread/INFO] [Sponge]: Block PLACE [RootCause: HunterzCZ][User: HunterzCZ][World: world][DimId: 0][OriginalState: minecraft:air][NewState: opencomputers:robot]
[14:32:16] [Server thread/INFO] [Sponge]: Tracking Block [RootCause: HunterzCZ][World: world(0)][Block: opencomputers:robot][Pos: BlockPos{x=-2, y=84, z=-1}]
[14:32:16] [Server thread/INFO] [Sponge]: Tracking Block [RootCause: HunterzCZ][World: world(0)][Block: opencomputers:robot][Pos: BlockPos{x=-2, y=84, z=-1}]
</pre>

Turn on:

<pre>
[14:33:04] [Server thread/INFO] [Sponge]: Tracking Block [RootCause: HunterzCZ][World: world(0)][Block: minecraft:bedrock][Pos: BlockPos{x=-2, y=83, z=-1}]
[14:33:04] [Server thread/INFO] [Sponge]: Tracking Block [RootCause: HunterzCZ][World: world(0)][Block: minecraft:air][Pos: BlockPos{x=-2, y=85, z=-1}]
[14:33:04] [Server thread/INFO] [Sponge]: Tracking Block [RootCause: HunterzCZ][World: world(0)][Block: minecraft:air][Pos: BlockPos{x=-2, y=84, z=-2}]
[14:33:04] [Server thread/INFO] [Sponge]: Tracking Block [RootCause: HunterzCZ][World: world(0)][Block: minecraft:air][Pos: BlockPos{x=-2, y=84, z=0}]
[14:33:04] [Server thread/INFO] [Sponge]: Tracking Block [RootCause: HunterzCZ][World: world(0)][Block: minecraft:air][Pos: BlockPos{x=-3, y=84, z=-1}]
[14:33:04] [Server thread/INFO] [Sponge]: Tracking Block [RootCause: HunterzCZ][World: world(0)][Block: minecraft:air][Pos: BlockPos{x=-1, y=84, z=-1}]
</pre>

Do robot.forward():

<pre>
[14:34:24] [Server thread/INFO] [Sponge]: Block PLACE [RootCause: net.minecraftforge.fml.common.eventhandler.ASMEventHandler][User: None][World: world][DimId: 0][OriginalState: minecraft:air][NewState: opencomputers:robot]
</pre>

</pre>

**Comment by ghost** [comment]
Still not fixed in the build 1971. Voting for this bug.

**Comment by inklit** [comment]
Confirmed it is still a problem on latest SpongeForge, @bloodmc said he had some insight on this issue a while back, any ideas?

**Comment by bloodmc** [comment]
The bug seems to be on OpenComputer's end. I filed a ticket here

https://github.com/MightyPirates/OpenComputers/issues/2173

**Comment by ghost** [comment]
I tried as workaround remove OC robots from block tracking:
<pre>
    }
    block-tracking {
        # Add block ids you wish to blacklist for player block placement tracking.
        block-blacklist=[
            "opencomputers:robot",
            "oc:robot",
            "oc:rs.bundledInput",
            "oc:rs.rednetInput",
            "oc:rs.bundledOutput"
        ]
        # If enabled, adds player tracking support for block positions. Note: This should only be disabled if you do not care who caused a block to change.
        enabled=true
    }
</pre>

But robots are still tracked...

**Comment by ghost** [comment]
Maybe will be useful some kind of entity/tile entity tracking blacklist too. (for debugging purposes).

**Comment by Vexatos** [comment]
https://github.com/MightyPirates/OpenComputers/pull/2174 should fix the issue with robots duplicating once it is merged.

**Comment by bloodmc** [comment]
I also added a patch which should fix issues with current versions of OpenComputers.

https://github.com/SpongePowered/SpongeCommon/commit/956f69ec03d62f96928d52b9ffdc5b046f70b483

Thanks again @Vexatos

**Comment by inklit** [comment]
Thanks a lot @bloodmc and @Vexatos! I can't wait until that fix is merged into 1.10.2.

**Comment by ghost** [comment]
Thanks a lot. Now robots move! Great job!

**Comment by bloodmc** [comment]
Fixed in latest OC and SpongeForge builds.
